internal/factsengine/policy.go

```diff
@@ -13,14 +14,22 @@ const (
 	FactsGatheringRequested = "Trento.Checks.V1.FactsGatheringRequested"
 )
 
-func (c *FactsEngine) handleEvent(_ string, request []byte) error {
+type eventHandler func(name string, event []byte) error
```

Comment:
question: why this new `makeEventHandler`?
was not enough to change the signature of of the `Listen` function in the `adapter.go` file to use the context?
I don't know, I find it more confusing than before.
Now, if we have other functions listening, the will need to copy this, which makes things more opaque in my opinion

Reply:
Because the listener in `FactsEngine.Listen` accepts a handling function with the signature `func(name string, event []byte) error`; and also, the context is coming from the scope not from the single event.

Reply:
Yes, of course, but you can change the signatures they way you prefer. That's not an issue
```
Listen(ctx context.Context, queue, exchange, routingKey string, handle func(ctx context.Context, contentType string, message []byte) error) error
```

Reply:
Idk, because then the `ctx` should be passed by `RabbitMQAdapter.Listen` which is not aware of it

Reply:
I personally would pass it.
Up to you @balanza , I will respect the decision you take :slightly_smiling_face: